doc/releases/changelog-dev.md

```diff
@@ -226,6 +226,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* `KerasLayer` and `TorchLayer` no longer mutate the input `QNode`'s interface.
+  [(#5800)](https://github.com/PennyLaneAI/pennylane/pull/5800)
```

Comment:
Should I also add an entry under "Breaking changes" since we now raise an error if the interface is not what we expect? cc @trbromley

Reply:
I would not consider this worthy of alerting anyone. Technically, everything breaks previous behaviour.  The question is whether or not anyone will care.  I cannot see anyone really caring about this change.

Reply:
Right, should be ok not to mention this. We can consider it a bugfix, and it's perhaps weird behaviour for the interface to be changed behind the scenes (but not raise an error).